hips/draw/simple.py

```diff
@@ -1,5 +1,111 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """HiPS tile drawing -- simple method."""
 
+import numpy as np
+from astropy.coordinates import SkyCoord
+from skimage import transform as tf
+from typing import List, Generator, Any
+
+from ..tiles import HipsSurveyProperties, HipsTile, HipsTileMeta
+from ..utils import WCSGeometry, compute_healpix_pixel_indices
+
 __all__ = [
+    'draw_sky_image',
+    'make_sky_image',
+    'SimpleTilePainter'
 ]
+
+
+def draw_sky_image(geometry: WCSGeometry, tiles: Generator[HipsTile, Any, Any]) -> np.ndarray:
+    """Draw sky image using the simple and quick method.
+
+    Parameters
+    ----------
+    geometry : `~hips.utils.WCSGeometry`
+        An object of WCSGeometry
+    tiles : List[HipsTile]
+        A list of HipsTile
+
+    Returns
+    -------
+    np.ndarray
+        Returns a numpy array containing all HiPS tiles projected onto it
+    """
+    all_sky = np.zeros(geometry.shape)
+    for tile in tiles:
+        draw_tile = SimpleTilePainter(geometry, tile)
+        all_sky += draw_tile.warp_image()
+    return all_sky
+
+
+class SimpleTilePainter:
+    """Paint a single tile using a simple projective transformation method.
+
+    The algorithm implemented is described here: :ref:`drawing_algo`.
+
+    Parameters
+    ----------
+    geometry : `~hips.utils.WCSGeometry`
+        An object of WCSGeometry
+    tile : `HipsTile`
+       An object of HipsTile
+    """
+
+    def __init__(self, geometry: WCSGeometry, tile: HipsTile) -> None:
+        self.geometry = geometry
+        self.tile = tile
+
+    def warp_image(self) -> np.ndarray:
+        """Warp a HiPS tile and a sky image"""
+        pt = self.tile.meta.apply_projection(self.geometry.wcs)
+        return tf.warp(self.tile.data.astype('float'), pt, output_shape=self.geometry.shape)
```

Comment:
Move the type conversion on a separate line.
Also note that http://scikit-image.org/docs/dev/user_guide/data_types.html says "never use astype" when using scikit-image. Please read that page and see if there's a better way.
In any case, please leave a comment saying why your changing the dtype here.